### PR TITLE
feat: convert link preview block to url text

### PR DESCRIPTION
**Pull request by LucasXu0**
### Feature Preview

closes https://github.com/AppFlowy-IO/AppFlowy/issues/4708

**desktop**
https://github.com/AppFlowy-IO/AppFlowy/assets/11863087/b2907bfd-bc61-432e-8ce8-8f07ae649393

**mobile**
Screenshot 2024-02-25 at 21 32 28

---

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/block_menu/block_menu_button.dart [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  final VoidCallback onTap;
+  final String tooltip;
+  final FlowySvgData iconData;
+
+  @override
+  Widget build(BuildContext context) {
+    return FlowyButton(
+      useIntrinsicWidth: true,
+      onTap: onTap,
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/block_menu/block_menu_button.dart:22]: Can make it nullable, then you don't have to pass `() {}` here https://github.com/AppFlowy-IO/AppFlowy/pull/4734/files#diff-fc379d4f0bf14d847cd3f81b8ce29be4fbf09893600d14c2fc60cfdbf0bccb06R152
```suggestion
    this.onTap,
  });

  final String tooltip;
  final FlowySvgData iconData;
  final VoidCallback? onTap;

  @override
  Widget build(BuildContext context) {
    return FlowyButton(
      useIntrinsicWidth: true,
      onTap: onTap,
```

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/link_preview/custom_link_preview.dart [hunk 2]
```diff
@@ -1,38 +1,50 @@
...
-        clipBehavior: Clip.hardEdge,
-        decoration: BoxDecoration(
-          border: Border.all(
-            color: Theme.of(context).colorScheme.onSurface,
-          ),
-          borderRadius: BorderRadius.circular(
-            6.0,
-          ),
+    final fontSize = PlatformExtension.isDesktopOrWeb ? 16.0 : 14.0;
+    final width = PlatformExtension.isDesktopOrWeb ? 180.0 : 120.0;
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/link_preview/custom_link_preview.dart:34]:
```suggestion
    final (fontSize, width) = PlatformExtension.isDesktopOrWeb ? (16.0, 180.0) : (14.0, 120.0);
```

Comment (Xazin) [reply frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/link_preview/custom_link_preview.dart:34]: Should the fontSize on Desktop follow the FontSize from Document Appearance?

Comment (LucasXu0) [reply frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/link_preview/custom_link_preview.dart]: Yes. Let me change it.

**Review by Xazin — APPROVED** [review]

**Pull request merged**